core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -666,23 +861,23 @@ private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOExcept
     // (2) the in stream reads from the local worker
     // (3) the in stream reads from a remote worker but there is no local worker
     boolean firstSeekOutsideFirstBlock =
-        mPos == 0 && mCurrentBlockInStream == null && !isInCurrentBlock;
+        mPositionState.getPos() == 0 && mCurrentBlockInStream == null && !isInCurrentBlock;
     if (!firstSeekOutsideFirstBlock && canCacheToLocalWorker()) {
       // Make sure that mCurrentBlockInStream and mCurrentCacheStream is updated.
       // mPos is not updated here.
       updateStreams();
```

Comment:
whenever you think the stream may be out of sync with the current position, and needs to be updated.

Reply:
ideally, whenever the position moves out of range of the current block, the streams are automatically updated in `mPositionState.increment` or `mPositionState.setPos`, but this affects the overall design of `FileInStream` since in the current code base, the logic for updating position (`mPos++`) and the logic for updating streams `updateStreams` are separated.